src/site/topic19.rst

```diff
@@ -245,6 +245,121 @@ Computational Complexity
 Fibonacci
 ---------
 
+* Consider the Fibonacci numbers
+* If you are not familiar with this sequence, see if you can figure out how it's created
+
+    :math:`0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 1597, 2584, 4181, 6765, ...`
+
+
+* Here's a hint
+
+    :math:`0, 1`
+
+    :math:`0, 1, 1`
+
+    :math:`0, 1, 1, 2`
+
+    :math:`0, 1, 1, 2, 3`
+
+    :math:`0, 1, 1, 2, 3, 5`
+
+    :math:`0, 1, 1, 2, 3, 5, 8`
+
+    :math:`0, 1, 1, 2, 3, 5, 8, 13`
+
+    :math:`\dots`
+
+
+* To generate this sequence, start with :math:`0, 1`, then to get the subsequent number, add the proceeding two together
+* Take a moment to think about how you would write an iterative method to generate these numbers
+    * An example is below --- make sure you understand this
+
+.. code-block:: java
+    :linenos:
+
+    static int iterativeFibonacci(int n) {
+        if (n == 0 || n == 1) {
+            return n;
+        }
+        int prev = 1;
+        int cur = 1;
+        int next = 0;
+        for (int i = 3; i <= n; i++) {
+            next = prev + cur;
+            prev = cur;
+            cur = next;
+        }
+        return next;
```

Comment:
This is wrong now. You're returning `f(2) = 0`. Something [like this](https://ideone.com/xFXWuP) works.

```java
if (n == 0) return 0;
int prev = 0, current = 1;
for (int i = 2; i <= n; ++i) {
    int next = prev + current;
    prev = current;
    current = next;
}
return current;
```

Reply:
copy/paste